Set up access privileges

kimberlylo opened the pull request:
Set up user permissions and how to check the user permissions in
ConversationHeader.java

kimberlylo committed:
Set up access privileges

Set up user permissions and how to check the user permissions in
ConversationHeader.java

markpilloff commented on src/codeu/chat/common/BasicController.java:
```diff
@@ -49,6 +49,6 @@
   //  representing the full state of the conversation on the server.
   //  Whether conversations can have the same title is undefined.
-  ConversationHeader newConversation(String title, Uuid owner);
+  ConversationHeader newConversation(String title, Uuid owner, Integer defaultByte);
```
Likely no need for the "Integer" class here (and almost everywhere else --- except when it's the parameter of a generic as in HashMap<..., Integer>). Just a plain old "int" should suffice.

markpilloff replied:
Also (here and elsewhere) defaultByte isn't a very informative name. defaultPermission (that's what this is, right?) would be much clearer.

markpilloff commented on src/codeu/chat/common/ConversationHeader.java:
```diff
@@ -44,23 +46,88 @@ public ConversationHeader read(InputStream in) throws IOException {
   public final Time creation;
   public final String title;
+  public final Integer defaultByte;
+  public final HashMap<Uuid, Integer> userLevels;
```
Just to be clear, it's perfectly fine to use "Integer" here (in fact, you must) and still store/retrieve "int" values. Java is a quirky language.

markpilloff commented on src/codeu/chat/common/ConversationHeader.java:
```diff
@@ -44,23 +46,88 @@ public ConversationHeader read(InputStream in) throws IOException {
+  // owner == 2
+  // creator == 3
+  
+  public enum permissions {
```
This is a nice enum but it doesn't really get used (except very briefly in the ConversationHeader constructor where you access creator.getPermissionLevel() (aka, "3"). There's definitely potential cleanliness to using an enum, but given that almost everywhere else you just work with the raw integer value, I would suggest in the interest of time / simplicity that you just stick with the integer representation.

markpilloff reviewed (approved):
Nice work Kimberly. I've added a few comments that should be easy fixes. Overall looks very good so feel free to merge after making those changes.

kimberlylo committed:
Changed Integer use, renamed defaultByte, removed enum

Channged Integer use to int use, reanmed defaultByte to
defaultPermissions, removed enum

kimberlylo committed:
Merge branch 'master' into ConversationAccessControlPrivileges
